### PR TITLE
Feature/templates config

### PR DESCRIPTION
**Pull request by Kaydayo**
## Description

This PR introduces the integration of third-party APIs/SDKs to enable interaction with external services as part of our system. It includes SDK connections for slack, allowing our application to retrieve data , authorise, send notifications and manage system bot

### Todos

- [x] Tested and working locally
- [ ] Unit tests added (if appropriate)
- [ ] Code changes documented
- [x] Requested review from >=1 devs on the team

### How to test

[Add detailed steps to test the changes that have been made]

### Screenshots and/or Gifs

[Insert screenshots and/or gifs showing the graphic representation of the change]

### Associated JIRA Tasks

- [Task Code: Task Description](http://jira_task_link.com)

### Known Issues

[Describe any known issue with this change]

### CODE REVIEW

File: src/base/base-company/base-company.service.ts [hunk 1]
```diff
@@ -149,7 +149,7 @@ export class BaseCompanyService extends CrudService<
         where: { id },
       },
     );
-
+    console.log(companyRequest,"SHO RIIRAN")
```
Comment (ayopedro) [inline src/base/base-company/base-company.service.ts]: remove logs

File: src/common/third-party/providers/slack/slack-integration.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      throw new Error(error?.message || 'an error occurred creating config')
+    }
+  }
+  async connect(config: SlackConfig): Promise<WebClient> {
+    const client = new WebClient(config.slackAccessToken);
+    return client;
+  }
+
+  getIntegrationUri(payload?: Record<string, any>): string {
+    const authBaseUri = 'https://slack.com/oauth/v2/authorize';
```
Comment (ayopedro) [inline src/common/third-party/providers/slack/slack-integration.ts:49]: store this in the env

File: src/company/integrations/integrations.service.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    };
+  }
+
+  private getIntegration(
+    integration_type: IntegrationProviders,
+  ): BaseIntegrationProvider {
+    const getPlatform = this.getIntegrations();
+    const provider = getPlatform[integration_type];
+    if (!provider) {
+      throw new Error(`Integration provider not found for ${integration_type}`);
```
Comment (ayopedro) [inline src/company/integrations/integrations.service.ts:35]: throw nest formatted error for consistency sake

**Review by ayopedro — CHANGES_REQUESTED** [review]
also format codes.

File: src/base/base-company/base-company.service.ts (hunk 1 above)
Comment (Kaydayo) [reply src/base/base-company/base-company.service.ts]: create a hook for this please

File: src/common/third-party/providers/slack/slack-integration.ts (hunk 2 above)
Comment (Kaydayo) [reply src/common/third-party/providers/slack/slack-integration.ts:49]: noted

File: src/company/integrations/integrations.service.ts (hunk 3 above)
Comment (Kaydayo) [reply src/company/integrations/integrations.service.ts:35]: okie

File: src/base/base-company/base-company.service.ts (hunk 1 above)
Comment (ayopedro) [reply src/base/base-company/base-company.service.ts]: a hook?

File: src/common/config/app.config.ts [hunk 4]
```diff
@@ -113,6 +113,9 @@ const appConfig = {
     user: {
       scope: env('SLACK_USER_SCOPE'),
     },
+    base: {
+      url: env('SLACK_BASE_URL'),
+    },
```
Comment (ayopedro) [inline src/common/config/app.config.ts:118]:
```suggestion
    slack: {
      base_url: env('SLACK_BASE_URL'),
    },
```

File: src/company/integrations/integrations.service.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
```
Comment (ayopedro) [inline src/company/integrations/integrations.service.ts:35]:
```suggestion
      throw new NotFoundException(`Integration provider not found for ${integration_type}`);
```

File: src/company/integrations/integrations.service.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+          },
+        });
+        result = true;
+      }
+    } catch (error) {
+      result = false;
+      throw new HttpException(
+        error.message || 'an error occurred',
+        HttpStatus.UNPROCESSABLE_ENTITY,
+      );
```
Comment (ayopedro) [inline src/company/integrations/integrations.service.ts:68]:
```suggestion
      throw new UnprocessableEntityException(
        error.message || 'an error occurred'
      );
```

**Review by ayopedro — APPROVED** [review]

**Review by ayopedro — APPROVED** [review]

**Pull request merged**